imap_processing/codice/codice_l1a.py

```diff
@@ -725,6 +856,99 @@ def log_dataset_info(datasets: dict[int, xr.Dataset]) -> None:
+    ----------
+    packets : xarray.Dataset
+        Dataset containing the packets, needed to determine priority order
+        and data quality.
+    decompressed_data : list[list[int]]
+        The decompressed data to reshape, in the format <epoch>[<priority>[<event>]].
+    num_priorities : int
+        The number of priorities in the data product (differs between CoDICE-Lo
+        and CoDICE-Hi).
+
+    Returns
+    -------
+    data : dict[str, numpy.ndarray]
+        The reshaped, CDF-ready arrays. The keys of the dictionary represent the
+        CDF variable names, and the values represent the data.
+    """
+    # Dictionary to hold all the (soon to be restructured) direct event data
+    data: dict[str, np.ndarray] = {}
+
+    # Determine the number of epochs to help with data array initialization
+    # There is one epoch per set of priorities
+    num_epochs = len(packets.epoch.data) // num_priorities
+
+    # Initialize data arrays for each priority and field to store the data
+    # We also need arrays to hold number of events and data quality
+    for priority_num in range(num_priorities):
+        for field in constants.LO_DE_BIT_STRUCTURE:
+            if field not in ["Priority", "Spare"]:
+                data[f"P{priority_num}_{field}"] = np.full(
+                    (num_epochs, 10000), 255, dtype=np.uint16
+                )
+        data[f"P{priority_num}_NumEvents"] = np.full(num_epochs, 255, dtype=np.uint16)
+        data[f"P{priority_num}_DataQuality"] = np.full(num_epochs, 255, dtype=np.uint16)
+
+    # decompressed_data is one large list of values of length
+    # (<number of epochs> * <8 priorities>)
+    # Chunk the data into each epoch
+    for epoch_index in range(num_epochs):
+        # Determine the starting and ending indices of the epoch
+        epoch_start = epoch_index * num_priorities
+        epoch_end = epoch_start + num_priorities
+
+        # Extract the data for the epoch
+        epoch_data = decompressed_data[epoch_start:epoch_end]
+
+        # The order of the priorities and data quality flags are unique to each
+        # epoch and can be gathered from the packet data
+        priority_order = packets.priority[epoch_start:epoch_end].data
+        data_quality = packets.suspect[epoch_start:epoch_end].data
+
+        # For each epoch/priority combo, iterate over each event
+        for i, priority_num in enumerate(priority_order):
+            priority_data = epoch_data[i]
+
+            # Number of events and data quality can be determined at this stage
+            num_events = len(priority_data) // num_priorities
+            data[f"P{priority_num}_NumEvents"][epoch_index] = num_events
+            data[f"P{priority_num}_DataQuality"][epoch_index] = data_quality[i]
+
+            # Iterate over each event
+            for event_index in range(num_events):
+                event_start = event_index * num_priorities
+                event_end = (event_index + 1) * num_priorities
```

Comment:
Earlier you did start + num_priorities, so for consistency it might be good to use that notation here too.
```suggestion
                event_end = event_start + num_priorities
```